maintenance/code-backports.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# silence debug output from VCS
+beQuiet="--quiet"
+#beQuiet=""
+version=`basename "$1" | sed s/squid-//`
+vnext=$(( $version + 1 ))
+srcbranch=v$vnext
+
+cd $1
+
+# Dev branch does not receive backports
+test -f .BASE && exit 0
+
+# clean the current workspace
+gitCleanWorkspace ()
+{
+  git clean $beQuiet -xdf --exclude="\.BASE"
+  git checkout $beQuiet -- .
+}
+
+if ! test -d .git ; then
+  echo "ERROR: missing git repository"
+  exit 1;
+fi
+if ! test -d ../squid-$vnext ; then
+  echo "ERROR: missing git repository to backport from"
+  exit 1;
+fi
+if test -e ../squid-$vnext/.BASE ; then
+  srcbranch=`cat ../squid-$vnext/.BASE`
+fi
+
+echo "FOUDN srcbranch=$srcbranch"
+
+# Prepare backports branch for updates
+git checkout $beQuiet v$version-next-backports 2>/dev/null &&
+  gitCleanWorkspace &&
+    git rebase origin/v$version 2>/dev/null || ( git rebase --abort 2>/dev/null ; exit 1 )
```

Comment:
I do not think `exit` will have the desired effect in this particular use case. It will _not_ stop `code-backports.sh` execution AFAICT. Parenthesis like that launch a subshell, and exit exits from that subshell. Please test and adjust as needed.

----

For example, the following script:

```sh
false || (echo failed; exit 2);
echo success!
```

produces:

```
failed
success!
```

and exits with code 0.

Reply:
replaced